rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala

```diff
@@ -1014,6 +1014,41 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
     errorLog.readAndClearErrorVector should be(Vector.empty[InterpreterError])
   }
 
+  it should "substitute before serialization" in {
+    implicit val errorLog = new ErrorLog()
+    implicit val costAccounting =
+      CostAccountingAlg.unsafe[Task](CostAccount.zero)
+    val splitRand = rand.splitByte(0)
+    val unsubProc: Par =
+      New(bindCount = 1, p = EVar(BoundVar(1)), locallyFree = BitSet(0))
+    val subProc: Par =
+      New(bindCount = 1, p = GPrivate("zero"), locallyFree = BitSet())
+    val serializedProcess         = subProc.toByteString
+    val toByteArrayCall: Par      = EMethod("toByteArray", unsubProc, List[Par]())
+    def wrapWithSend(p: Par): Par = Send(Quote(GString("result")), List[Par](p), false, BitSet())
+
+    val result = withTestSpace { space =>
```

Comment:
I guess it'd be much easier to write such a test by writing inline rholang in a string literal and asserting execution results. I'd gladly work on infrastructure for writing such tests.

Reply:
We could even easily convert selected tests to 'rholang source' style by simply pretty-printing the input source. So it'd not only be a long term investment - it'd give immediate returns in terms of test code readability.

Reply:
I've said this before:
No.
That would literally break the point of these tests, which is to test some portion of the interpreter **without** depending on the other parts.

Reply:
I remember you said this. I'm bringing this because it seemed to me you asked if we have sth similar in one of the discord channels a couple days ago. I guess it was about sth else / for a different purpose then.